extensions/stackdriver/stackdriver.cc

```diff
@@ -152,31 +163,19 @@ FilterDataStatus StackdriverContext::onResponseBody(size_t body_buffer_length,
   return FilterDataStatus::Continue;
 }
 
-StackdriverRootContext *StackdriverContext::getRootContext() {
-  RootContext *root = this->root();
-  return dynamic_cast<StackdriverRootContext *>(root);
+StackdriverRootContext* StackdriverContext::getRootContext() {
+  RootContext* root = this->root();
+  return dynamic_cast<StackdriverRootContext*>(root);
 }
 
 void StackdriverContext::onLog() {
-  bool isOutbound = getRootContext()->isOutbound();
+  auto* root = getRootContext();
+  bool isOutbound = root->isOutbound();
```

Comment:
You can follow the stats filter here and move most of this login into the RootContext code.